### PR TITLE
Added unit test for RedisCacheWrapper.

### LINKED ISSUE #16761: Add Unit Test for RedisCacheWrapper shouldn't implement IDisposable
Watch the last Orchard Core meeting and see @sebastienros comment about this

Related to #16720

### PR DESCRIPTION
**Pull request by gvkries**
Fixes #16761

/cc @hishamco

### CODE REVIEW

**Comment by hishamco** [comment]
I think I will do it but it's great that you picked up 😁

**Comment by hishamco** [comment]
I will review you the PR tomorrow if not today

**Comment by gvkries** [comment]
> I think I will do it but it's great that you picked up 😁

Yeah, I saw that you're assigned to that issue. I was just thinking about how one may test something this and wanted to share my approach. Feel free to close this PR, if you have a different or better way of doing it.🤗

File: src/OrchardCore.Modules/OrchardCore.Redis/OrchardCore.Redis.csproj [hunk 1]
```diff
@@ -19,4 +19,10 @@
     <ProjectReference Include="..\..\OrchardCore\OrchardCore.Redis.Abstractions\OrchardCore.Redis.Abstractions.csproj" />
   </ItemGroup>
 
+  <ItemGroup>
+    <AssemblyAttribute Include="System.Runtime.CompilerServices.InternalsVisibleTo">
+      <_Parameter1>OrchardCore.Tests</_Parameter1>
+    </AssemblyAttribute>
+  </ItemGroup>
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Redis/OrchardCore.Redis.csproj:26]:
```suggestion
  <ItemGroup>
    <InternalsVisibleTo Include="OrchardCore.Tests" />
  </ItemGroup>
```

File: src/OrchardCore.Modules/OrchardCore.Redis/Services/RedisCacheWrapper.cs [hunk 2]
```diff
@@ -11,21 +10,21 @@ namespace OrchardCore.Redis.Services;
 /// <remarks>
 /// This is done by not disposing the `RedisCache` instance which would otherwise close the redis connection.
 /// </remarks>
-public sealed class RedisCacheWrapper : IDistributedCache
+internal sealed class RedisCacheWrapper : IDistributedCache
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Redis/Services/RedisCacheWrapper.cs:13]: This is a breaking change, I might agree to set it `internal` if we have an extension method to add redis service

File: test/OrchardCore.Tests/Modules/OrchardCore.Redis/RedisCacheWrapperTests.cs [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+public class RedisCacheWrapperTests
+{
+    [Fact]
+    public void RedisCacheWrapperMustNotDisposeInnerCache()
+    {
+        var redisCache = new Mock<IDistributedCache>();
+        var disposable = redisCache.As<IDisposable>();
+
+        object testObject = new RedisCacheWrapper(redisCache.Object);
```
Comment (hishamco) [inline test/OrchardCore.Tests/Modules/OrchardCore.Redis/RedisCacheWrapperTests.cs:14]:
```suggestion
        var testObject = new RedisCacheWrapper(redisCache.Object);
```

**Review by hishamco — CHANGES_REQUESTED** [review]

File: test/OrchardCore.Tests/Modules/OrchardCore.Redis/RedisCacheWrapperTests.cs (hunk 3 above)
Comment (gvkries) [reply test/OrchardCore.Tests/Modules/OrchardCore.Redis/RedisCacheWrapperTests.cs:14]: This must be an object explicitly, as `RedisCacheWrapper` currently doesn't implement `IDisposable`.

File: src/OrchardCore.Modules/OrchardCore.Redis/Services/RedisCacheWrapper.cs (hunk 2 above)
Comment (gvkries) [reply src/OrchardCore.Modules/OrchardCore.Redis/Services/RedisCacheWrapper.cs:13]: It is, but this type should have never been public, it should be resolved through DI only.

Comment (hishamco) [reply src/OrchardCore.Modules/OrchardCore.Redis/Services/RedisCacheWrapper.cs:13]: Forget my comment, while the class resides in the module

File: src/OrchardCore.Modules/OrchardCore.Redis/Services/RedisCacheWrapper.cs [hunk 4]
```diff
@@ -11,21 +10,21 @@ namespace OrchardCore.Redis.Services;
...
 {
-    private readonly RedisCache _cache;
+    private readonly IDistributedCache _cache;
 
-    public RedisCacheWrapper(IOptions<RedisCacheOptions> optionsAccessor) => _cache = new RedisCache(optionsAccessor);
+    public RedisCacheWrapper(IDistributedCache cache) => _cache = cache ?? throw new ArgumentNullException(nameof(cache));
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Redis/Services/RedisCacheWrapper.cs:17]: I'm not sure why you changed the parameter type?

Comment (gvkries) [reply src/OrchardCore.Modules/OrchardCore.Redis/Services/RedisCacheWrapper.cs:17]: Because `RedisCache` cannot be easily mocked for unit testing.

**Review by hishamco — APPROVED** [review]

**Comment by hishamco** [comment]
@sebastienros do you have anything else to add here?

**Review by sebastienros — APPROVED** [review]

**Pull request merged**